plugins/plugin-chart-echarts/src/BoxPlot/transformProps.ts

```diff
@@ -0,0 +1,178 @@
+ * under the License.
+ */
+import {
+  CategoricalColorNamespace,
+  ChartProps,
+  convertMetric,
+  DataRecord,
+  getNumberFormatter,
+  QueryFormDataMetric,
+} from '@superset-ui/core';
+import { BoxPlotQueryFormData } from './types';
+import { EchartsProps } from '../types';
+import { extractGroupbyLabel } from '../utils/series';
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queryData } = chartProps;
+  const data: DataRecord[] = queryData.data || [];
+  const {
+    colorScheme,
+    groupby = [],
+    metrics: formDataMetrics,
+    numberFormat,
+    xTicksLayout,
+  } = formData as BoxPlotQueryFormData;
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+  const numberFormatter = getNumberFormatter(numberFormat);
+
+  // TODO: remove cast once metrics is cleaned up in QueryFormData
+  const metrics = (formDataMetrics as QueryFormDataMetric[]).map(
+    metric => convertMetric(metric).label,
+  );
+
+  const transformedData = data
+    .map(datum => {
+      return metrics.map(metric => {
+        const groupbyLabel = extractGroupbyLabel({ datum, groupby });
+        const name = metrics.length === 1 ? groupbyLabel : `${groupbyLabel}, ${metric}`;
+        return {
+          name,
+          value: [
+            datum[`${metric}__low`],
+            datum[`${metric}__q1`],
+            datum[`${metric}__median`],
+            datum[`${metric}__q3`],
+            datum[`${metric}__high`],
+            datum[`${metric}__mean`],
+            datum[`${metric}__count`],
+            datum[`${metric}__outliers`],
+          ],
+          itemStyle: {
+            color: colorFn(groupbyLabel),
+            opacity: 0.6,
+            borderColor: colorFn(groupbyLabel),
+          },
+        };
+      });
+    })
+    .flatMap(row => row);
+
+  const outlierData = data
+    .map(datum => {
+      return metrics.map(metric => {
+        const groupbyLabel = extractGroupbyLabel({ datum, groupby });
+        const name = metrics.length === 1 ? groupbyLabel : `${groupbyLabel}, ${metric}`;
+        // Outlier data is a nested array of numbers (uncommon, therefore no need to add to DataRecordValue)
+        const outlierDatum = (datum[`${metric}__outliers`] || []) as number[];
+        return {
+          name: 'outlier',
+          type: 'scatter',
+          data: outlierDatum.map(val => [name, val]),
+          tooltip: {
+            formatter: (param: { data: [string, number] }) => {
+              const headline = `<p><strong>${param.data[0]}</strong></p>`;
+              const stats = `${numberFormatter(param.data[1])}`;
+              return headline + stats;
+            },
+          },
+          itemStyle: {
+            color: colorFn(groupbyLabel),
+          },
+        };
+      });
+    })
+    .flat(2);
+
+  let axisLabel;
+  if (xTicksLayout === '45°') axisLabel = { rotate: -45 };
```

Comment:
I always feel like I _should_ suggest `switch` in cases like this instead of `if else` but this seems plenty easy to read, and efficient enough.

Reply:
My python background forbids me from writing switches 🐍